rust/crd/src/ldap.rs

```diff
@@ -0,0 +1,201 @@
+use std::collections::BTreeMap;
+
+use stackable_operator::commons::authentication::AuthenticationClassProvider;
+use stackable_operator::commons::ldap::LdapAuthenticationProvider;
+
+use crate::authentication::ResolvedAuthenticationClasses;
+
+const DEFAULT_LDAP_PORT: u16 = 1389;
+const DEFAULT_LDAP_TLS_PORT: u16 = 1636;
+
+#[derive(Clone, Debug)]
+pub struct DruidLdapSettings {
+    pub ldap: LdapAuthenticationProvider,
+}
+
+impl DruidLdapSettings {
+    pub fn new_from(
+        resolved_authentication_config: ResolvedAuthenticationClasses,
+    ) -> Option<DruidLdapSettings> {
+        let maybe_authentication_class =
+            resolved_authentication_config.get_ldap_authentication_class();
+
+        if let Some(authentication_class) = maybe_authentication_class {
+            if let AuthenticationClassProvider::Ldap(ref provider) =
+                authentication_class.spec.provider
+            {
+                return Some(DruidLdapSettings {
+                    ldap: provider.clone(),
+                });
+            }
+        }
+        None
+    }
+
+    fn add_druid_system_authenticator_config(&self, config: &mut BTreeMap<String, Option<String>>) {
+        const PREFIX: &str = "druid.auth.authenticator.DruidSystemAuthenticator";
+
+        config.insert(format!("{PREFIX}.type"), Some("basic".to_string()));
```

Comment:
FYI there is also https://crates.io/crates/const_format
Not sure if we want to use it though

Reply:
Interesting to know, thanks for sharing! I would be fine with leaving it as it is, as there does not seem to be enough upside vs adding an additional dependency in this particular case.